modules/gdscript/gdscript_cache.cpp

```diff
@@ -173,11 +174,16 @@ Ref<GDScript> GDScriptCache::get_shallow_script(const String &p_path, const Stri
 		return singleton->shallow_gdscript_cache[p_path];
 	}
 
+	Ref<GDScriptParserRef> parser_ref = get_parser(p_path, GDScriptParserRef::PARSED, r_error);
+	if (r_error != OK) {
+		return Ref<GDScript>();
+	}
```

Comment:
This is not equivalent to the previous logic. Now it seems that if the parsing fails, the script will not be cached by `shallow_gdscript_cache` (the previous logic is only an invalid script will report an error, this will hardly appear, may it appear in multi-threading?)